jenkinsfiles/android_jenkinsfile.groovy

```diff
@@ -1,13 +1,40 @@
 /**
 * Android Jenkinsfile
 */
-node('android') {
-    stage 'Checkout'
+node("android"){
+  stage("checkout"){
     checkout scm
+  }
 
-    stage 'Build'
-    sh "./gradlew clean assembleDebug"
+  stage ("Prepare"){
+    writeFile file: 'app/src/main/assets/fhconfig.properties', text: params.FH_CONFIG_CONTENT
+  }
 
-    stage 'Archive'
-    archiveArtifacts artifacts: 'app/build/outputs/apk/*.apk', excludes: 'app/build/outputs/apk/*-unaligned.apk'
-}
+  stage("build"){
+    sh 'chmod +x ./gradlew'
+    sh './gradlew clean assembleRelease' // builds app/build/outputs/app-release-unsigned.apk file
+  }
+
+  stage("sign"){
+    if (params.BUILD_CONFIG == 'release'){
+        signAndroidApks (
+            keyStoreId: "${params.BUILD_CREDENTIAL_ID}",
+            keyAlias: "${params.BUILD_CREDENTIAL_ALIAS}",
+            apksToSign: "**/*-unsigned.apk",
+            // uncomment the following line to output the signed APK to a separate directory as described above
+            // signedApkMapping: [ $class: UnsignedApkBuilderDirMapping ],
+            // uncomment the following line to output the signed APK as a sibling of the unsigned APK, as described above, or just omit signedApkMapping
+            // you can override these within the script if necessary
+            // androidHome: '/usr/local/Cellar/android-sdk'
+        )
+    }
```

Comment:
probably print out a message to say that the default developer signing key is used for debug builds